bin/make-vps.sh

```diff
@@ -62,21 +70,32 @@ if [[ "${help}" -eq 1 ]] ; then
   exit
 fi
 
-if [[ -z "${target_name}" ]] ; then
-  usage
-  exit 1
-fi
- 
-target_test=$(gnt-instance list "${target_name}" 2> /dev/null | tail -n+2 | awk '{print $1}')
-if [[ -n "${target_test}" ]] ; then
-  echo "ERROR: '${target_name}' already exists"
-  usage
-  exit 1
-fi
+while [[ -z "${target_name}" ]] ; do
+  read -p "Enter the VPS hostname to create: " target_name
+  target_test=$(gnt-instance list "${target_name}" 2> /dev/null | tail -n+2 | awk '{print $1}')
+  if [[ -n "${target_test}" ]] ; then
+    unset target_name
+    echo "ERROR: '${target_name}' already exists"
+  fi
+done
 
-target_ip=$(getent ahostsv4 "${target_name}" | awk '{print $1}' | head -n1)
+while [[ -z "${target_ip}" ]] ; do
+  read -p "Enter the VPS IP address to use: " target_ip
+  if [[ ! "$target_ip" =~ ^(([1-9]?[0-9]|1[0-9][0-9]|2([0-4][0-9]|5[0-5]))\.){3}([1-9]?[0-9]|1[0-9][0-9]|2([0-4][0-9]|5[0-5]))$ ]]; then
+    echo "ERROR: $target_ip isn't a valid IP address"
+    continue
+  fi
+  if grep $target_ip /etc/hosts; then
+    echo "There is an existing entry for this IP in /etc/hosts."
+    echo "This likely indicates that it previously belonged to a different VPS that was torn down but not cleaned up or that the IP database is out of date."
+    echo "Please remove this IP from /etc/hosts if it is indeed free before continuing"
+    exit 1
```

Comment:
This is not an error if you're re-installing an old instance. This needs to check if the old IP entry, as well as the hostname for that entry and allow if they match.

Reply:
Ah, I didn't realize a re-install of an existing instance with the tool was possible. Sorry bout that

Reply:
Not directly, but by doing `gnt-instance remove <foo>` then installing again.

Reply:
What would be a good fix to both accommodate the use case you're describing and stay safe from leftover `/etc/hosts` entries?